src/services/utilities.ts

```diff
@@ -2735,9 +2735,7 @@ namespace ts {
 
         type PackageJsonRaw = Record<typeof dependencyKeys[number], Record<string, string> | undefined>;
         const dependencyKeys = ["dependencies", "devDependencies", "optionalDependencies", "peerDependencies"] as const;
-        const stringContent = host.readFile(fileName);
-        if (!stringContent) return undefined;

```

Comment:
This was the problem. Non-empty package.jsons with parse errors were already handled, but empty files were not.